launcher/modplatform/flame/FlamePackIndex.cpp

```diff
@@ -65,15 +65,7 @@ void Flame::loadIndexedPackVersions(Flame::IndexedPack& pack, QJsonArray& arr)
         // pick the latest version supported
         file.mcVersion = versionArray[0].toString();
         file.version = Json::requireString(version, "displayName");
-        file.fileName = Json::requireString(version, "fileName");
-        file.downloadUrl = Json::ensureString(version, "downloadUrl");
-        if(file.downloadUrl.isEmpty()){
-            //FIXME : HACK, MAY NOT WORK FOR LONG
-            file.downloadUrl = QString("https://media.forgecdn.net/files/%1/%2/%3")
-                    .arg(QString::number(QString::number(file.fileId).leftRef(4).toInt())
-                            ,QString::number(QString::number(file.fileId).rightRef(3).toInt())
-                            ,QUrl::toPercentEncoding(file.fileName));
-        }
+        file.downloadUrl = Json::requireString(version, "downloadUrl");
```

Comment:
Can't this crash if CF doesn't provide a download URL for the pack? I think it's best to keep it as `ensureString` just in case :^)

Reply:
Is it fine like this?

Reply:
Would be cool if we had an filter option to show all modpacks, even if they don't allow third-party redistribuiton, instead of just not showing them at all. But that's a separate issue! I think it's fine like that :)

Reply:
It actually still shows the modpacks. It just doesn't offer any versions to download

Reply:
Oh I read it wrong then sorry :sweat_smile: 
It's very fine then!!!